assets/js/modules/analytics/datastore/form.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Analytics form datastore.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import invariant from 'invariant';
+
+// Actions
+const SET_FORM = 'SET_FORM';
+
+export const INITIAL_STATE = {};
+
+export const actions = {
+	setForm( formID, data ) {
+		invariant( formID, 'formID is required for updating data.' );
+		invariant( typeof data === 'object', 'data must be an object to merge.' );
+
+		return {
+			payload: { formID, data },
+			type: SET_FORM,
+		};
+	},
+};
+
+export const controls = {};
+
+export const reducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case SET_FORM:
+			const { formID, data } = payload;
+			const formKey = `form::${ formID }`;
```

Comment:
Any reason for the key prefix?

Reply:
So a `formID` can't conflict with other state in the store. We could put that in the formID itself, but it seemed a bit safer this way.